data/game_data/main.py

```diff
@@ -0,0 +1,309 @@
+    """
+    Creates the parameters directory along with the readme file and the empty
+    main.csv file that will hold all investigated parameters
+    """
+    directory = pathlib.Path("data/parameters/")
+    directory.mkdir(parents=True, exist_ok=True)
+    readme_contents = (
+        "# Parameters"
+        "\n\nThis directory keeps track of all the parameters investigated so far. The"
+        "\ncontents of `main.csv` correspond to parameter values of the model in the"
+        "\nfollowing order:\n"
+    )
+    with (directory / "README.md").open("w") as file:
+        file.write(readme_contents)
+        file.write("".join("\n\t" + key + "," for key, _ in problem_parameters.items()))
+    if not (directory / "main.csv").exists():
+        write_data_to_csv(sorted(problem_parameters.keys()))
+
+
+def generate_data_for_current_parameters(processes, **problem_parameters):
+    """
+    Generates the routing matrix, the payoff matrix for the row player (A) and
+    the payoff matrix for the column player (B), given a set of parameters.
+
+    Returns
+    -------
+    numpy array
+    numpy array
+    numpy array
+    """
+    payoff_matrix_A, payoff_matrix_B, routing_matrix = abg.game.get_payoff_matrices(
+        processes=processes,
+        **problem_parameters,
+    )
+    return routing_matrix, payoff_matrix_A, payoff_matrix_B
+
+
+def write_README_for_current_parameters_directory(readme_path, **problem_parameters):
+    """
+    Writes the readme file (README.md) for the directory of the given set of
+    problem parameters.
+
+    Parameters
+    ----------
+    readme_path : pathlib.Path object
+        the path where the readme file will be located
+    """
+    parameters_string = "".join(
+        "\n\t" + key + " = " + str(value) for key, value in problem_parameters.items()
+    )
+    readme_contents = (
+        "# Experiments for game"
+        "\n\nThis directory consists of the data for the following set of parameters: \n"
+        + "".join(parameters_string)
+        + "\n\nThe directory is structured in the following way:\n\n"
+        "\t|-- main.csv\n"
+        "\t|-- README.md\n"
+        "\t|-- routing\n"
+        "\t|   |-- main.csv\n"
+        "\t|   |-- README.md \n"
+        "\t|-- A\n"
+        "\t|   |-- main.csv\n"
+        "\t|   |-- README.md\n"
+        "\t|-- B\n"
+        "\t|   |-- main.csv\n"
+        "\t|   |-- README.md\n"
+    )
+    with readme_path.open("w") as file:
+        file.write(readme_contents)
+
+
+def write_README_for_current_parameters_sub_directories(readme_path, output_name):
+    """
+    Writes the readme files (README.md) for the sub-directories routing, A or B
+
+    Parameters
+    ----------
+    readme_path : pathlib.Path object
+        the path that the readme will go on
+    output_name : string
+        the numpy that we write the readme for, i.e. "routing matrix",
+        "payoff matrix of A", "payoff matrix of B"
```

Comment:
Can we add instructions to read in the data (https://numpy.org/doc/stable/reference/generated/numpy.loadtxt.html#numpy.loadtxt`), note that if you end up going the way of using that other tool to write all the arrays to a single file then those instructions can be at the higher level.